### PR TITLE
feat: SPARQL resource shape lookup

### PR DESCRIPTION
**Pull request by tpluscode**
Docs PR: https://github.com/zazuko/data-centric.zazuko.com/pull/28

### CODE REVIEW

File: packages/core/query/resourceShapes.rq [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  } UNION {
+    ?resourceShape sh:targetNode ?subject .
+  }
+
+  optional {
+    ?resourceShape sh:property/sh:path ?property .
+    ?subject ?property ?object .
+  }
+
+  VALUES ?resource { sh:this }
```
Comment (tpluscode) [inline packages/core/query/resourceShapes.rq:22]: I was surprised to find that this `VALUES` only work inside the query and not outside. Thus, the `sh:this` is replaced with the request URL

File: packages/core/test/support/in-memory-clients.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import type * as Oxigraph from 'oxigraph'
```
Comment (tpluscode) [inline packages/core/test/support/in-memory-clients.ts:1]: This modules exports im-memory oxigraph wrapped to match the interface of `sparql-http-client`

File: packages/core/test/lib/resourceShape.test.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import type { KopflosConfig } from '../../lib/Kopflos.js'
+import Kopflos from '../../lib/Kopflos.js'
+import inMemoryClients from '../support/in-memory-clients.js'
+import { ex } from '../support/ns.js'
+import { createStore } from '../support/testData.js'
+
+describe('lib/resourceShape', () => {
+  let options: KopflosConfig
+
+  beforeEach(createStore(import.meta.url, { format: 'trig' }))
```
Comment (tpluscode) [inline packages/core/test/lib/resourceShape.test.ts:13]: Loads RDF from `resourceShape.test.ts.trig` and sets mocha `graph` and `store` properties to the mocha context.

The contents of the store/graph will be taken from a graph constructed from the test `describe/context` titles

File: packages/core/query/resourceShapes.rq (hunk 1 above)
Comment (giacomociti) [reply packages/core/query/resourceShapes.rq:22]: indeed, the query looks non-trivial, and I'm a bit concerned about its speed with large APIs. What about getting rid of `VALUES` and placing `sh:this` directly in all the needed places?

Comment (tpluscode) [reply packages/core/query/resourceShapes.rq:22]: Since it's doing replacement, might also do it that way.

Let's address this when it becomes a problem.

**Pull request merged**